options.go

```diff
@@ -611,6 +620,41 @@ func parseJwtIssuers(issuers []string, msgs []string) ([]jwtIssuer, []string) {
 	return parsedIssuers, msgs
 }
 
+func parseIPWhitelist(ipWhitelistStrs []string, msgs []string) ([]*net.IPNet, []string) {
+	parsedIPNets := make([]*net.IPNet, 0, len(ipWhitelistStrs))
+
+	for ipIndex, ipStr := range ipWhitelistStrs {
+		if !strings.ContainsRune(ipStr, '/') {
+			ip := net.ParseIP(ipStr)
+			if ip == nil {
+				msgs = append(msgs, fmt.Sprintf(
+					"ip_whitelist[%d] (%s) looks like a IP address, but could not be recognized", ipIndex, ipStr))
+				continue
+			}
+
+			// nolint:gocritic
+			if ip.To4() != nil {
+				ipStr += "/32"
+			} else if ip.To16() != nil {
+				ipStr += "/128"
+			} else {
+				msgs = append(msgs, fmt.Sprintf("ip_whitelist[%d] (%s) address neither IPv4 or IPv6", ipIndex, ipStr))
+				continue
+			}
+		}
+
+		_, ipNet, err := net.ParseCIDR(ipStr)
+		if err != nil {
+			msgs = append(msgs, fmt.Sprintf("ip_whitelist[%d] (%s) can't parse as CIDR: %s", ipIndex, ipStr, err.Error()))
```

Comment:
Use `%v` for errors, has the same effect but is more idiomatic
```suggestion
			msgs = append(msgs, fmt.Sprintf("whitelisted IP (%q) can't parse as CIDR: %v", ipStr, err))
```

Reply:
Haha same catch you made before, sorry I ported this code from my old branch before I got the first review on my other PR.  I should've gone through and made all the relevant suggestions in the other PR.